src/smpp_socket.erl

```diff
@@ -373,11 +431,13 @@ handle_pkt(Pkt, binding, State) ->
     case handle_bind_req(Pkt, State) of
         {ok, bound, State1} ->
             State2 = set_keepalive_timeout(State1, smsc),
-            {ok, bound, dequeue_req(State2)};
+            {ok, bound, State2};
         Other ->
             Other
     end;
-handle_pkt(Pkt, bound, State) ->
+handle_pkt(Pkt, StateName, State) when StateName == bound;
+                                       StateName == rate_limit_cooldown;
+                                       StateName == in_flight_limit_cooldown ->
     case handle_request(Pkt, State) of
```

Comment:
There is a hardcode below:
```erlang
handle_request(Pkt, State) ->
    report_unexpected_pkt(Pkt, bound, State),
```
It looks line we need to pass `StateName` to `handle_request` now.

Reply:
fixed